engine/rpcserver.go

```diff
@@ -1020,9 +1020,52 @@ func (s *RPCServer) CancelOrder(_ context.Context, r *gctrpc.CancelOrderRequest)
 		Data: fmt.Sprintf("order %s cancelled", r.OrderId)}, nil
 }
 
+// CancelBatchOrders cancels an orders specified by exchange, currency pair and asset type
+func (s *RPCServer) CancelBatchOrders(_ context.Context, r *gctrpc.CancelBatchOrdersRequest) (*gctrpc.CancelBatchOrdersResponse, error) {
+	exch := s.GetExchangeByName(r.Exchange)
+	if exch == nil {
+		return nil, errExchangeNotLoaded
+	}
+
+	p, err := currency.NewPairFromStrings(r.Pair.Base, r.Pair.Quote)
+	if err != nil {
+		return nil, err
+	}
+
+	a, err := asset.New(r.AssetType)
+	if err != nil {
+		return nil, err
+	}
+
+	err = exch.CancelOrder(&order.Cancel{
+		AccountID:     r.AccountId,
+		ID:            r.OrdersId,
+		Side:          order.Side(r.Side),
+		WalletAddress: r.WalletAddress,
+		Pair:          p,
+		AssetType:     a,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return &gctrpc.CancelBatchOrdersResponse{}, nil
+}
+
 // CancelAllOrders cancels all orders, filterable by exchange
 func (s *RPCServer) CancelAllOrders(_ context.Context, r *gctrpc.CancelAllOrdersRequest) (*gctrpc.CancelAllOrdersResponse, error) {
-	return &gctrpc.CancelAllOrdersResponse{}, common.ErrNotYetImplemented
+	exch := s.GetExchangeByName(r.Exchange)
+	if exch == nil {
+		return &gctrpc.CancelAllOrdersResponse{}, errExchangeNotLoaded
+	}
+
+	resp, err := exch.CancelAllOrders(nil)
```

Comment:
optional - It would be better to create an `order.Cancel{}` with exchange, and optionally pair and asset if provided. Exchanges like FTX require these, so this function isn't fully implemented, but I understand its out of scope for what you have developed.